src/reconcile/rec_write.c

```diff
@@ -1965,7 +1965,7 @@ __rec_compression_adjust(WT_SESSION_IMPL *session, uint32_t max, size_t compress
      *	Writing a shared memory location without a lock and letting it
      * race, minor trickiness so we only read and write the value once.
      */
-    WT_ORDERED_READ(current, *adjustp);
+    WT_READ_ONCE(current, *adjustp);
```

Comment:
This is a good example of how `WT_READ_ONCE` makes reviewing the code more complex. With `WT_ORDERED_READ`, I know that no read of a shared variable in future code could happen before the read here into `current`, so I don't need to understand the following code to check whether such a thing is feasible. i.e: Are there other shared variables involved? If so would it matter if the read happened before this read.

In this case, I think the change is OK. The only other shared variable in play is `btree->maxmempage_image`, and I happen to know that the value for that isn't regularly changed, so we don't need to care if the read from it happened prior to the read into `current`.

Reply:
Are orderings like the one between `btree->maxmempage_image` and `current` something we need to keep front of mind when reading any and all code? What I like about `READ_ONCE` here is it tells the reader there's no need to think about ordering. If there was then we'd use a barrier and add a comment explaining exactly why the barrier was introduced. 

My goal with changes like this is to remove ambiguous ordering constraints wherever possible. I'm happy to keep this an `ORDERED_READ` and mention in the comment why we'd like to have ordering here. That would still resolve the current issue where we have ordering constraints in this code but can't explain why or whether it's intentional.